pages/architecture/ccip/tutorial.mdx

```diff
@@ -87,6 +93,30 @@ fn resolve(
     }
 ```
 
+### URL Management
+
+Within our contract we will add methods to management our api URIs. You can define different functions to add, remove and retrieve URIs.
+
+```rust
+fn get_uris(self: @TContractState) -> Array<felt252>;
+fn add_uri(ref self: TContractState, new_uri: Span<felt252>);
+fn remove_uri(ref self: TContractState, index: felt252);
+```
+
+It is crucial to emit an event of type `StarknetIDOffChainResolverUpdate` whenever a URI is added or removed. Starknet ID uses this event to discover your resolver and track any changes associated with it. This allows Starknet ID to resolver offchain resolver in its api through the endpoint [`domain_to_addr`](https://docs.starknet.id/devs/starknetid-api#domain-to-address).
+
+Here is the structure of the event:
+
+```rust
+#[derive(Drop, starknet::Event)]
+struct StarknetIDOffChainResolverUpdate {
+    uri_added: Span<felt252>,
+    uri_removed: Span<felt252>,
+}
+```
+
+Please note that the Starknet ID fetches new `StarknetIDOffChainResolverUpdate` events at an interval of every 10 minutes.
```

Comment:
I think you missed a word here. Didn't you mean "the Starknet ID indexer"?

Reply:
ah yes indeed !